src/main/java/com/pinterest/secor/main/ConsumerMain.java

```diff
@@ -54,6 +55,9 @@ public static void main(String[] args) {
         }
         try {
             SecorConfig config = SecorConfig.load();
+            if (!config.getUploadOnShutdown()) {
```

Comment:
Only adding this hook, if uploadOnShutdown is disabled, otherwise we cannot be sure that deletion will happen after upload. Java does not provide strong order for shutdown hooks, they can be executed in any order & in parallel.

Other option would be implementing some kind of "shutdown hook registry" combining all hooks into one thread and registring all Runnables as one shutdown hook. But I think we do not need that, since files are deleted after upload anyways.

Reply:
Take a look at DeleteOnExitHook.java, there seems a way to register your hook as the last one.  If ordering cannot be done, I think you can modify the hook in uploadOnShutdown to have them call your hook's folder deletion methods.  It's OK to have both hooks to call delete folders twice.